### DISCUSSION TITLE
Should I use WikiLinks with or without path?

### DISCUSSION THREAD

**Discussion by stella10x1** [zk-org/zk#311] [Help]
Hello, in my [previous question](https://github.com/mickael-menu/zk/discussions/310) I asked about how to access notes by ID in subfolders from cli. There @mickael-menu stated:

> Are you using `[[WikiLink]]` or `[markdown](link)` with Neovim? If `[[WikiLink]]`, that explains the discrepancy. `zk list` takes paths but a `[[WikiLink]]` is more flexible.

This brings me to the question should I use WikiLinks with or without paths? Does it even matter?

The background for this question is, that I wrote a (quick and dirty) [tool](https://codeberg.org/lingling/nb-to-zk-importer) to import my notes from [nb](https://github.com/xwmx/nb) to [zk](https://github.com/mickael-menu/zk). That tool keeps track of the old note IDs from [nb](https://github.com/xwmx/nb) and the new note IDs from [zk](https://github.com/mickael-menu/zk). In the end it replaces all [nb links](https://github.com/xwmx/nb#-linking) in all notes with the correct [zk](https://github.com/mickael-menu/zk) links. One step was to [use a glob pattern](https://codeberg.org/lingling/nb-to-zk-importer/src/commit/1fc702494f6c53f733dbeafaa75f73b55218d44a/src/correct_links.py#L21) to find the correct path. I now wonder if I can skip that step and just address my notes among each other with only the ID.

I know that it is possible, because it does work in NeoVim. With only using the ID there is an obvious advantage, that I get the flexibility to move around the note in folders without correcting links in other notes. **Are there any disadvantages in only using the ID?**

**Comment by mickael-menu** [comment]
Unfortunately there's no official specification for what a WikiLink is in Markdown and many apps implement slightly different rules.

WikiLinks in `zk` must be any unique portion of a note path, so for `dir/a8f3 Note title.md`, all these are valid:

* `[[dir/a8f3 Note title]]`
    * Most accurate, easy to transform into actual Markdown links or do post-processing.
    * Fragile, if you rename a note or move it to a different directory.
* `[[Note title]]` (if no other note with the same title)
    * Most readable.
    * Fragile, if you rename a note.
* `[[a8f3]]`
    * Least readable.
    * Resilient to changes, you can move or rename the note.

So it's a matter of trade-off. You might also want to look at the other apps you want to use with your notebook (Obsidian, Neuron/Emanote, etc.) to find a convention that is most interoperable.

Personally, I'm uncomfortable with the loose spec of the WikiLinks. In my notebook I use:

* No folders, all notes are located at the root of the notebook.
* The filenames are only a short 4-characters ID (1 679 616 unique IDs), e.g. `a8f3.md`.
    * I found that I have no use for seeing the note title in the filename, as I don't browse the notes from the file system. And I could always use `zk list -f "{{path}} {{title}}"`.
* I keep each note short and atomic, around a single idea or concept.
    * I'm following the [excellent examples from Andy Matuschak](https://notes.andymatuschak.org/z4SDCZQeRo4xFEQ8H4qrSqd68ucpgE6LU155C).
* In my notes, I use Markdown links for best interoperability (short filenames improve readability). When I can the note title is inlined into a paragraph to read flutenly.

```markdown
# Chronic insomnia is caused by anxiety

[Having trouble sleeping from time to time is normal](276q). A real chronic insomnia is diagnosed
as a few bad nights per week for more than 3 months.

Insomnia is a symptom of an overly active [fight-or-flight response](692z) system caused by
anxiety. As [being sleep-deprived increases anxiety](g2ki), insomnia is a vicious circle.

To break out of insomnia, do not resort to:

* Drugs, as [sleeping pills are harmful](5zij) and entertain the idea that you can't sleep on your own.
* Sleep hygiene advices, as they can generate health anxiety of their own [which is counterproductive](6s3f).

Instead, you need to fix the underlying emotional issues, for example by practicing [CBT-I](mnke).
```

```markdown
# Sleep hygiene advices can be counterproductive

There are plenty of sleep hygiene advices offered to people struggling with chronic insomnia.

However, such advices can have the perverse effect of putting too much emphasis on sleep, generating
health anxiety. This is counterproductive as [chronic insomnia is caused by anxiety](uavn).
```

(I've been insomnia free for a couple of years. Researching how it works actually helped me, who would have known 😄)
